### PR TITLE
feat(genqa): filter citations that points to same document

### PR DESCRIPTION
**Pull request by jelmedini**
[SVCC-4029](https://coveord.atlassian.net/browse/SVCC-4029)

- Filter citations that points to same documents.

Before:

Screenshot 2024-08-01 at 3 14 46 PM

After:
Screenshot 2024-08-02 at 3 11 39 PM

[SVCC-4029]: https://coveord.atlassian.net/browse/SVCC-4029?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by dmgauthier — APPROVED** [review]

**Comment by dmgauthier** [comment]
Should we test this ?

**Comment by louis-bompart** [comment]
> Should we test this ?

Yes. See other slice tests for inspiration ;)

**Review by lvu285 — APPROVED** [review]

**Review by louis-bompart — DISMISSED** [review]
UT

**Review by erocheleau — APPROVED** [review]
Just a suggestion if you don't need the flatMap?

File: packages/headless/src/features/generated-answer/generated-answer-slice.ts [hunk 1]
```diff
@@ -47,7 +47,12 @@ export const generatedAnswerReducer = createReducer(
       .addCase(updateCitations, (state, {payload}) => {
         state.isLoading = false;
         state.isStreaming = true;
-        state.citations = state.citations.concat(payload.citations);
+        const allCitations = state.citations.concat(payload.citations);
+        state.citations = allCitations.flatMap((citation, index, self) =>
+          self.findIndex((c) => c.uri === citation.uri) === index
+            ? [citation]
+            : []
+        );
```
[the remark this answers is not in the document]
Comment (louis-bompart) [reply packages/headless/src/features/generated-answer/generated-answer-slice.ts:55]:
```suggestion
        const citationMap = new Map<string, Citation>();
        for (const citationCollection of [state.citations, payload.citations]) {
          for (const citation of citationCollection) {
            citationMap.set(citation.uri, citation)
          }
        }
        state.citations = Array.from(citation.values())
```

**Review by fpbrault — APPROVED** [review]
LGTM!

**Pull request merged**